Build with -DCMAKE_BUILD_TYPE=Debug -DWITH_BFD=yes fails

This is the traceback I get
``` shell
/home/naveen/dev/symengine/symengine/utilities/teuchos/Teuchos_stacktrace.cpp: In function ‘void {anonymous}::process_section(bfd*, asection*, void*)’:
/home/naveen/dev/symengine/symengine/utilities/teuchos/Teuchos_stacktrace.cpp:193:8: error: ‘bfd_get_section_flags’ was not declared in this scope; did you mean ‘bfd_set_section_flags’?
  193 |   if ((bfd_get_section_flags(abfd, section) & SEC_ALLOC) == 0) {
      |        ^~~~~~~~~~~~~~~~~~~~~
      |        bfd_set_section_flags
/home/naveen/dev/symengine/symengine/utilities/teuchos/Teuchos_stacktrace.cpp:197:25: error: ‘bfd_get_section_vma’ was not declared in this scope; did you mean ‘bfd_set_section_vma’?
  197 |   bfd_vma section_vma = bfd_get_section_vma(abfd, section);
      |                         ^~~~~~~~~~~~~~~~~~~
      |                         bfd_set_section_vma
/home/naveen/dev/symengine/symengine/utilities/teuchos/Teuchos_stacktrace.cpp:203:49: error: cannot convert ‘bfd*’ to ‘const asection*’ {aka ‘const bfd_section*’}
  203 |   bfd_size_type section_size = bfd_section_size(abfd, section);
      |                                                 ^~~~
      |                                                 |
      |                                                 bfd*
In file included from /home/naveen/dev/symengine/symengine/utilities/teuchos/Teuchos_stacktrace.cpp:73:
/usr/include/bfd.h:1206:35: note:   initializing argument 1 of ‘bfd_size_type bfd_section_size(const asection*)’
 1206 | bfd_section_size (const asection *sec)
      |                   ~~~~~~~~~~~~~~~~^~~
make[2]: *** [symengine/utilities/teuchos/CMakeFiles/teuchos.dir/build.make:128: symengine/utilities/teuchos/CMakeFiles/teuchos.dir/Teuchos_stacktrace.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:784: symengine/utilities/teuchos/CMakeFiles/teuchos.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```
This is on Ubuntu 20.04, with binutils-dev v2.34-6ubuntu1. Compiling without these options works.

--- a/symengine/utilities/teuchos/Teuchos_stacktrace.cpp
+++ b/symengine/utilities/teuchos/Teuchos_stacktrace.cpp
@@ -208,8 +208,8 @@ void process_section(bfd *abfd, asection *section, void *_data)
     return;
   }
 
-#ifdef bfd_get_section_size
-  bfd_size_type section_size = bfd_get_section_size(abfd, section);
+#ifdef bfd_section_size
+  bfd_size_type section_size = bfd_section_size(abfd, section);
 #else
   bfd_size_type section_size = bfd_section_size(section);
 #endif

--- a/symengine/utilities/teuchos/Teuchos_stacktrace.cpp
+++ b/symengine/utilities/teuchos/Teuchos_stacktrace.cpp
@@ -190,17 +190,29 @@ void process_section(bfd *abfd, asection *section, void *_data)
     // If we already found the line, exit
     return;
   }
+#ifdef bfd_get_section_flags
   if ((bfd_get_section_flags(abfd, section) & SEC_ALLOC) == 0) {
-    return;
-  }
+#else
+  if ((bfd_section_flags(section) & SEC_ALLOC) == 0) {
+#endif
+  return;
+}
 
+#ifdef bfd_get_section_vma
   bfd_vma section_vma = bfd_get_section_vma(abfd, section);
+#else
+  bfd_vma section_vma = bfd_section_vma(section);
+#endif
   if (data->addr < section_vma) {
     // If the addr lies above the section, exit
     return;
   }
 
-  bfd_size_type section_size = bfd_section_size(abfd, section);
+#ifdef bfd_get_section_size
+  bfd_size_type section_size = bfd_get_section_size(abfd, section);
+#else
+  bfd_size_type section_size = bfd_section_size(section);
+#endif
   if (data->addr >= section_vma + section_size) {
     // If the addr lies below the section, exit
     return;

